### PR TITLE
Vibhansa/block upload of unchanged file

### PR DESCRIPTION
**Pull request by vibhansa-msft**
Do not upload file unless it has been modified.
User can choose this behavior using "--upload-if-modified=true" cli option.

### CODE REVIEW

**Review by NaraVen — COMMENTED** [review]
We also need a new test case for this. Even the nightly build will do.  We have to mock the PUT operation somehow.

File: blobfuse/blobfuse.cpp [hunk 1]
```diff
@@ -49,6 +49,7 @@ const struct fuse_opt option_spec[] =
     OPTION("--max-concurrency=%s", concurrency),
     OPTION("--cache-size-mb=%s", cache_size_mb),
     OPTION("--empty-dir-check=%s", empty_dir_check),
+    OPTION("--upload-if-modified=%s", upload_if_modified),
```
[the remark this answers is not in the document]
Comment (vibhansa-msft) [reply blobfuse/blobfuse.cpp:52]: corrected

**Review by NaraVen — APPROVED** [review]
the only risk is if metadata or permissions or properties are modified

**Comment by vibhansa-msft** [comment]
Only list api is blocked so if application tries to access any file immediately after load, it will still work. Objective was to only block the list call to stop implicit listing done by libfuse.

**Pull request merged**